### ISSUE TITLE
Solarflow current Battery temperature is missing via MQTT

### ISSUE THREAD

**Issue by AzraelsDisk** [Zendure/developer-device-data-report#31]
In the Zendure app, the temperature of each battery is shown, but this information is missing on MQTT. Only a max temp is provided, but this is not the current one. 
![image](https://github.com/Zendure/developer-device-data-report/assets/117444481/26f28415-5319-4368-8526-a6065001b2e4)

**Comment by AzraelsDisk** [comment]
Oh, I haven’t seen this…. That’s really disappointing…Am 15.11.2023 um 01:38 schrieb nerra0pos ***@***.***>:﻿
True, unfortunately, Zendure has apparently abandoned any further development. Very disappointing.

**Comment by Zendure** [comment]
The battery pack information has been updated. Developers can perform data parsing on the packData object as needed. Field explanation:

socLevel: Battery level
maxTemp: Battery temperature
maxVol: Maximum battery voltage
minVol: Minimum battery voltage
sn: Battery serial number

**Comment by AzraelsDisk** [comment]
Yes but is max temp the current one? Or the max of the day?Am 15.11.2023 um 11:49 schrieb Zendure ***@***.***>:﻿
The battery pack information has been updated. Developers can perform data parsing on the packData object as needed. Field explanation:
socLevel: Battery level
maxTemp: Battery temperature
maxVol: Maximum battery voltage
minVol: Minimum battery voltage
sn: Battery serial number

**Comment by Zendure** [comment]
maxTemp refers to the current highest temperature of the battery cells inside the battery pack. It represents the present temperature rather than the maximum temperature recorded throughout the day.

**Comment by AzraelsDisk** [comment]
Thanks for the explanation, but why as an example is on the mqtt 29 degrees reported and in the app 5,9 degrees? Am 16.11.2023 um 02:49 schrieb Zendure ***@***.***>:﻿
maxTemp refers to the current highest temperature of the battery cells inside the battery pack. It represents the present temperature rather than the maximum temperature recorded throughout the day.

**Comment by Zendure** [comment]
It seems like there might be a discrepancy between the temperature values reported on MQTT and the app. The maxTemp reported on MQTT is in Kelvin, so you would need to convert it to Celsius or Fahrenheit according to your preference.

**Comment by ragman11** [comment]
Is it possible to create a sensor for the maxTemp, to extract it from the packData?

**Comment by z-master42** [comment]
Yes, that's possible.

**Comment by ragman11** [comment]
and how the entry look like in the mqtt.yaml. I tried, but could not find a solution.

**Comment by z-master42** [comment]
I have a concrete idea, but have not yet been able to implement and test it due to a lack of time. Hopefully within the next few days.

**Comment by ragman11** [comment]
thank you for the code, works perfect

**Comment by z-master42** [comment]
Is explained under Steps to Subscribe to Device Data.

**Comment by z-master42** [comment]
Maybe you have to force it. When I click on the battery icon in the SolarFlow overview in the Zendure app so that it shows the charge level and temperature of the individual batteries, all MQTT values are updated, even those that have not yet been transmitted.

**Comment by z-master42** [comment]
As I wrote in my previous post, my MQTT values (including everything that comes via packData) are initially updated when I go to the battery detail view in the app.

**Comment by leow149** [comment]
Hi, is there any news on this since it is not closed? Seems like at least the SF 800 Pro still only reports the "hyper" temperature instead of the individual battery pack temperatures

**Comment by z-master42** [comment]
I don't think you'll get any response here. Doesn't the Pro have native MQTT?

**Comment by leow149** [comment]
> I don't think you'll get any response here. Doesn't the Pro have native MQTT?

It does. Will I get more data from there vs from the cloud?

**Comment by z-master42** [comment]
As far as I know, yes. In addition, there would also be the official Home Assistant integration.
